sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/ChatCompletionsOptions.java

```diff
@@ -952,7 +978,95 @@ public static ChatCompletionsOptions fromJson(JsonReader jsonReader) throws IOEx
             deserializedChatCompletionsOptions.responseFormat = responseFormat;
             deserializedChatCompletionsOptions.tools = tools;
             deserializedChatCompletionsOptions.toolChoice = toolChoice;
+            deserializedChatCompletionsOptions.parallelToolCalls = parallelToolCalls;
             return deserializedChatCompletionsOptions;
         });
     }
+
+    /*
+     * An upper bound for the number of tokens that can be generated for a completion, including visible output tokens
+     * and reasoning tokens.
+     */
+    @Generated
+    private Integer maxCompletionTokens;
+
+    /*
+     * Options for streaming response. Only set this when you set `stream: true`.
+     */
+    @Generated
+    private ChatCompletionStreamOptions streamOptions;
+
+    /*
+     * Whether to enable parallel function calling during tool use.
+     */
+    @Generated
+    private Boolean parallelToolCalls;
+
+    /**
+     * Get the maxCompletionTokens property: An upper bound for the number of tokens that can be generated for a
+     * completion, including visible output tokens and reasoning tokens.
+     *
+     * @return the maxCompletionTokens value.
+     */
+    @Generated
+    public Integer getMaxCompletionTokens() {
+        return this.maxCompletionTokens;
+    }
+
+    /**
+     * Set the maxCompletionTokens property: An upper bound for the number of tokens that can be generated for a
+     * completion, including visible output tokens and reasoning tokens.
+     *
+     * @param maxCompletionTokens the maxCompletionTokens value to set.
+     * @return the ChatCompletionsOptions object itself.
+     */
+    @Generated
+    public ChatCompletionsOptions setMaxCompletionTokens(Integer maxCompletionTokens) {
+        this.maxCompletionTokens = maxCompletionTokens;
+        return this;
+    }
+
+    /**
+     * Get the streamOptions property: Options for streaming response. Only set this when you set `stream: true`.
+     *
+     * @return the streamOptions value.
+     */
+    @Generated
+    public ChatCompletionStreamOptions getStreamOptions() {
+        return this.streamOptions;
+    }
+
+    /**
+     * Set the streamOptions property: Options for streaming response. Only set this when you set `stream: true`.
+     *
+     * @param streamOptions the streamOptions value to set.
+     * @return the ChatCompletionsOptions object itself.
+     */
+    @Generated
+    public ChatCompletionsOptions setStreamOptions(ChatCompletionStreamOptions streamOptions) {
```

Comment:
Since we have separate streaming methods, should we have a separate options types for streaming and non-streaming methods?

Reply:
official OpenAI API has share the same option type.
https://github.com/openai/openai-openapi/blob/master/openapi.yaml#L12978

Reply:
offline discussion: Create an overload method to take CompletionStreamOption and ChatCompletionStreamOption instead,